cmd/template-resolver/utils/templateResolver.go

```diff
@@ -58,28 +69,21 @@ func HandleFile(yamlFile string) ([]byte, error) {
 	return yamlBytes, nil
 }
 
-// ProcessTemplate takes a YAML byte array input, unmarshals it to a Policy, processes the
-// templates, and marshals it back to YAML, returning the resulting byte array. Validation is
-// performed along the way, returning an error if any failures are found. It uses the
-// `hubKubeConfigPath` and `clusterName` to establish a dynamic client with the hub to resolve any
-// hub templates it finds.
+// ProcessTemplate takes a YAML byte array input, unmarshals it to a Policy, ConfigPolicy,
+// or object-templates-raw, processes the templates, and marshals it back to YAML,
+// returning the resulting byte array. Validation is performed along the way, returning
+// an error if any failures are found. It uses the `hubKubeConfigPath` and `clusterName`
+// to establish a dynamic client with the hub to resolve any hub templates it finds.
+// Note that a hub resolver only supports resolving Policies.
 func ProcessTemplate(yamlBytes []byte, hubKubeConfigPath, clusterName string) ([]byte, error) {
 	policy := unstructured.Unstructured{}
+	var err error
```

Comment:
Declaring the variable here isn't needed. You can stick with `err := yaml.Unmarshal(...)`